### DISCUSSION TITLE
Allow commenter to unresolve comments

### DISCUSSION THREAD

**Discussion by daviball** [community/community#21003] [Pull Requests]
Commenters who are not the PR owner and do not have write permissions to the repo cannot mark comments as Resolved, which is fair enough.  However, if a comment has been marked Resolved by someone else, such commenters are unable to mark it as Unresolved.  The original commenter should always be able to mark a resolved comment as Unresolved if they do not believe it has been resolved, even if they do not have write permission to the repo.

**Comment by fdwr** [comment]
> Commenters who are not the PR owner and do not have write permissions to the repo cannot mark comments as Resolved, which is fair enough.

🤔 This feels really bizarro, as the creator of the comment should have permission to resolve their own comment, and there's even a feature request (https://github.com/orgs/community/discussions/12961) to make it so that *only* the commenter can resolve it, once they feel it's resolved. I agree though that the commenter should at a minimum be able to unresolve a comment.

**Reply by daviball** [reply]
Agree, in many situations it would be even better if only the commenter could mark comments as resolved.

**Reply by fdwr** [reply]
> Agree, in many situations it would be even better if only the commenter could mark comments as resolved.

That could be especially useful with junior devs where reviewers want to ensure comments have actually been addressed. I miss ADO's comment state where reviewer or author can set a comment to:

- Active - comment just created
- Pending - author actively working on it (often times you can skip this step straight to resolved, but it's helpful info if the comment is more involved and requires some time after the next iteration).
- Resolved - addressed as far as the author can tell, which the reviewer can then close
- Won't fix - author chose to not do it
- Closed - the reviewer closes it after inspecting the next iteration, either seeing it was properly resolved or conceding to the "won't fix".

Granted, *most* of the time comments were either just *active*/*resolved*/*won't fix* (with *pending* and *closed* being used less often), but it enabled more issue clarity.
